lib/probes/http.js

```diff
@@ -333,35 +339,37 @@ function patchServer (module, protocol) {
           }
         }
 
-        // if an x-trace-options header set x-trace-options-response header
-        // if there is an x-trace we need to handle the response as oboe doesn't generate
-        // appopriate responses for those cases.
+        // if there is an x-trace-options header set an x-trace-options-response header. if there
+        // is an x-trace we need to handle the response as oboe doesn't generate appopriate responses
+        // for those cases. in that case if the authStatus is OK then fill out the rest the response.
         if (xtraceOpts) {
           const responseParts = [];
 
-          if (!ao.cfg.triggerTraceEnabled) {
-            responseParts.push('trigger-trace=trigger-trace-disabled');
-          } else if (xtrace) {
-            responseParts.push(`trigger-trace=${ttRequested ? 'ignored' : 'not-requested'}`);
-            addKeysToKVPairs();
-          } else {
-            // only add auth key to the response header if an x-trace-options-signature header
-            if (xtraceOptsSig) {
-              responseParts.push(`auth=${settings.authMessage}`);
+          // if there is an auth message we always generate a reply.
+          if (xtraceOptsSig) {
+            responseParts.push(`auth=${settings.authMessage}`);
+          }
+
+          // if there is an x-trace header then oboe does not generate the message. if there is not an
+          // x-trace header then oboe's message should be right.
+          if (xtrace) {
+            if (settings.typeProvisioned !== 0) {
+              log.warn(`x-trace with x-trace-options provisioned as ${settings.typeProvisioned} authStatus = ${settings.authStatus}`);
             }
-            // if the auth status is "good" add other trigger trace response items.
             if (settings.authStatus <= 0) {
-              // set the options-response header to either trigger-trace status code or not requested
-              responseParts.push(`trigger-trace=${ttRequested ? settings.message : 'not-requested'}`)
-              if (ttRequested) {
-                kvpairs.TriggeredTrace = true;
-              }
-              addKeysToKVPairs();
-              // finally add any ignored keys to the response header
-              const ignored = Object.keys(xtraceOptsHash).join(',');
-              if (ignored) {
-                responseParts.push(`ignored=${ignored}`)
-              }
+              responseParts.push(`trigger-trace=${ttRequested ? 'ignored' : 'not-requested'}`);
+            }
+            addKeysToKVPairs();
+          } else if (settings.authStatus <= 0) {
+            responseParts.push(`trigger-trace=${ttRequested ? settings.message : 'not-requested'}`)
+            if (ttRequested) {
```

Comment:
would `if(settings.typeProvisioned == 1)` be more correct in this case?

Reply:
if there isn't an xtrace then a different type can't be provisioned now. (maybe in the future in which case i probably need to change the whole sequence to make a decision based on the requested-provisioned pair.)

but for now it's really standing for `trigger-trace` is present in a request and is used to make a decision about which message to display.